### ISSUE TITLE
Allow running as root

### ISSUE THREAD

**Issue by sheerun** [pgbouncer/pgbouncer#244]
I have issue with PgBouncer as it refuses to start with "PgBouncer should not run as root".

I'm running pgbouncer inside Docker image so it's perfectly normal to run as a root..

**Comment by PJMODOS** [comment]
No, we don't want that. Nothing stops you from running as non-root in docker. You have to do that with postgres anyway.

**Comment by sheerun** [comment]
I run pgbouncer along app, not postgres (via heroku-buildpack-pgbouncer).

The buildpacks are executes as root because otherwise it would result in duplication of files in layers (when you chmod file in docker, it duplicates this file in new layer...).

Generally running app as root is required in my use case, couldn't pgbouncer allow at least for a flag that changes this behavior?

**Comment by minhnhatnoe** [comment]
Hi. I'm running PgBouncer in Docker as part of my application. In my setup, Postgres runs outside of Docker and exposes a Unix socket with Peer Authentication that I need to connect to.

My solution is to mount /run/postgres inside the container, then make PgBouncer accepts the Unix socket as usual. Since I'm using rootless Docker, 'root' inside the Docker container is mapped to the triggering user. Therefore, to make PgBouncer able to Peer Authenticate into Postgres with the triggering user, I must be able to run PgBouncer as `root`.

Now that Docker is a much more popular solution for deploying application, letting people run PgBouncer as root seems like a sensible thing to do.

**Comment by JelteF** [comment]
I don't understand why you cannot run as another user in the docker container in your situation. You can use the `USER` dockerfile command to change the user to a non-root user. For instance this is done by the bitnami container:
https://github.com/bitnami/containers/blob/bfce99ed7c60e0cc9ca10a632b677b0240e39a95/bitnami/pgbouncer/1/debian-12/Dockerfile#L67C1-L67C10

**Comment by minhnhatnoe** [comment]
Thanks for the quick response!

Let my host user (the user outside of Docker) be `abc`. Since it's rootless Docker, UID 0 in my Docker container is mapped to `abc`, and no other user in the Docker container has this mapping.

I'm trying to interact with Postgres using a Unix socket. When PgBouncer starts connecting to Postgres through the socket that I mounted, Postgres will attempt to verify PgBouncer's identity by calling getpeereid ([peer authentication](https://www.postgresql.org/docs/current/auth-peer.html)). If PgBouncer's UID is 0 in the Docker container, it will be mapped to `abc`'s UID, and Peer Authentication will succeed.

I have tried both using the `USER` Dockerfile directive and the runtime option `--user` of `docker run`, and they don't seem to work. I received peer authentication failure when trying to connect using `psql` to the Unix socket exposed by PgBouncer, but succeed when connecting using `psql` to the Postgres' Unix socket mounted inside the container. I ran `psql` as root, while PgBouncer was running as `postgres` (a user not mapped to the host user namespace).
